src/Bicep.Core.IntegrationTests/ScenarioTests.cs

```diff
@@ -5298,4 +5298,71 @@ public void Test_Issue11981()
             ("BCP037", DiagnosticLevel.Error, "The property \"baz\" is not allowed on objects of type \"{ bar: string }\". No other properties are allowed."),
         });
     }
+
+    // https://github.com/Azure/bicep/issues/12417
+    [TestMethod]
+    public void Test_Issue12417()
+    {
+        var result = CompilationHelper.CompileParams(
+("parameters.bicepparam", @"
+using 'test.bicep'
+
+param azureEnvironments
+
+param region = azureEnvironments.AzureCloud.eventHubName.dev
+"),
+("test.bicep", @"
+@description('Azure environment specific configuration settings')
+type AzureEnvironmentType = {
+  @description('Azure  US Government specific settings')
+  AzureUSGovernment: {
+    @description('Azure  Gov Event Hub ResourceIds')
+    eventHubName: {
+      prod: 'resourceId'
+      qa: 'resourceId'
+      dev: 'resourceId'
+    }
+    @description('Azure  Gov supported regions')
+    region: {
+      usgovvirginia: 'ugv'
+      usgovtexas: 'ugt'
+    }
+    @description('Azure  Gov Log Analytics ResourceIds')
+    workspace: {
+      prod: 'resourceId'
+      qa: 'resourceId'
+      dev: 'resourceId'
+    }
+  }
+  @description('Azure  Global specific settings')
+  AzureCloud: {
+    @description('Azure  Global Event Hub ResourceIds')
+    eventHubName: {
+      prod: 'resourceId'
+      qa: 'resourceId'
+      dev: 'resourceId'
+    }
+    @description('Azure  Global supported regions')
+    region: {
+      eastus: 'eus'
+      westus: 'wus'
+    }
+    @description('Azure  Global Log Analytics ResourceIds')
+    workspace: {
+      prod: 'resourceId'
+      qa: 'resourceId'
+      dev: 'resourceId'
+    }
+  }
+}
+
+param azureEnvironments AzureEnvironmentType
```

Comment:
[nit]: You can simplify to `param azureEnvironments object` if you want the test to be smaller; the unhandled exception was caused by the incomplete `param azureEnvironments` statement in the parameters.bicepparam file.

Reply:
True - I personally like to just check in the closest to the exact sample provided in the issue report (within reason).